src/main/java/duke/tasklist/TaskList.java

```diff
@@ -0,0 +1,204 @@
+
+/**
+ * Represents a list of task within chat bot.
+ * Contains operations to add, delete and run task.
+ */
+public class TaskList {
+    private final List<Task> library;
+
+    /**
+     * Constructs TaskList Object.
+     */
+    public TaskList() {
+        this.library = new ArrayList<>(100);
+    }
+
+    /**
+     * Constructs TaskList Object.
+     * Loads task request from text file into list.
+     *
+     * @param taskFile scanner object of text file.
+     */
+    public TaskList(Scanner taskFile) throws DukeException{
+        this.library = new ArrayList<>(100);
+
+        // Extract lines
+        while(taskFile.hasNext()) {
+            String str = taskFile.nextLine();
+            add(str);
+        }
+        taskFile.close();
+    }
+
+    /**
+     * Returns task title of task request.
+     * Substrings input line after start word till end of string.
+     *
+     * @param input task request line.
+     * @param start substring after target word.
+     * @throws DukeException If start word doesn't exist.
+     * @return substring-ed title.
+     */
+    private String cut(String input, String start) throws DukeException {
+        String result;
+        try {
+            result = input.substring(input.indexOf(start) + start.length() + 1);
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new DukeException(String.format(Messages.EMPTY.toString(), start));
+        }
+        return result;
+    }
+
+    /**
+     * Returns task field of task request.
+     * Substrings input line between start word and end word.
+     *
+     * @param input task request line.
+     * @param start substring after target word.
+     * @param end substring before target word.
+     * @throws DukeException If start or end word doesn't exist.
+     * @return substring-ed task field.
+     */
+    private String cut(String input, String start, String end) throws DukeException {
+        String result;
+        try {
+            result = input.substring(input.indexOf(start) + start.length() + 1, input.indexOf(end));
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new DukeException(String.format(Messages.EMPTY.toString(), start));
+        }
+        return result;
+    }
+
+    /**
+     * Converts date time string into LocalDateTime.
+     * Parses date time by d/M/yyyy (optional HHmm).
+     * If optional is empty, default time is 2359.
+     *
+     * @param time date time string.
+     * @return datetime of date time string.
+     */
+    private LocalDateTime dateTime(String time) throws DukeException {
+        DateTimeFormatter fmt = new DateTimeFormatterBuilder()
+                .appendPattern("d/M/yyyy")
+                .optionalStart()
+                .appendPattern(" HHmm")
+                .optionalEnd()
+                .parseDefaulting(ChronoField.HOUR_OF_DAY, 23)
+                .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 59)
+                .toFormatter();
```

Comment:
indentation should be 8, seems like it's more than that

Reply:
It is 8 from the previous line,